From: Piotr Luc <piotr.luc@intel.com>
Date: Wed, 12 Oct 2016 20:26:34 +0200
Subject: perf/x86/intel: Add Knights Mill CPUID
Git-commit: 608284bf0def3ca5e6936920fcd84294101ef12d
Patch-mainline: v4.9-rc2
References: FATE#321922
Signed-off-by: Tony Jones <tonyj@suse.de>

    [ suse: 
      adjust for context, no ef5f9f47d4ec4cf42bac48c7c4dafacc1b9f0630
       perf/x86/intel: Use Intel family macros for core perf events 

      adjust for e1069839dd6893d2135b2fc4d96e5d03d73c2c3d
      perf/x86: Move perf_event_intel.c => x86/events/intel/core.c
    ]

    perf/x86/intel: Add Knights Mill CPUID
    
    Add Knights Mill (KNM) to the list of CPUIDs supported by PMU.
    
    Signed-off-by: Piotr Luc <piotr.luc@intel.com>
    Reviewed-by: Dave Hansen <dave.hansen@intel.com>
    Cc: Andy Lutomirski <luto@kernel.org>
    Cc: Borislav Petkov <bp@alien8.de>
    Cc: Brian Gerst <brgerst@gmail.com>
    Cc: Denys Vlasenko <dvlasenk@redhat.com>
    Cc: H. Peter Anvin <hpa@zytor.com>
    Cc: Josh Poimboeuf <jpoimboe@redhat.com>
    Cc: Linus Torvalds <torvalds@linux-foundation.org>
    Cc: Peter Zijlstra <peterz@infradead.org>
    Cc: Thomas Gleixner <tglx@linutronix.de>
    Link: http://lkml.kernel.org/r/20161012182634.2462-1-piotr.luc@intel.com
    Signed-off-by: Ingo Molnar <mingo@kernel.org>

---
 arch/x86/kernel/cpu/perf_event_intel.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/arch/x86/kernel/cpu/perf_event_intel.c
+++ b/arch/x86/kernel/cpu/perf_event_intel.c
@@ -3775,6 +3775,7 @@ __init int intel_pmu_init(void)
 		break;
 
 	case 87: /* Knights Landing Xeon Phi */
+	case 133: /*  Knights Mill -- INTEL_FAM6_XEON_PHI_KNM */
 		memcpy(hw_cache_event_ids,
 		       slm_hw_cache_event_ids, sizeof(hw_cache_event_ids));
 		memcpy(hw_cache_extra_regs,
@@ -3789,7 +3790,7 @@ __init int intel_pmu_init(void)
 		x86_pmu.flags |= PMU_FL_HAS_RSP_1;
 		x86_pmu.flags |= PMU_FL_NO_HT_SHARING;
 
-		pr_cont("Knights Landing events, ");
+		pr_cont("Knights Landing/Mill events, ");
 		break;
 
 	case 142: /* 14nm Kabylake Mobile */
